### ISSUE TITLE
CVE-2019-11043: add PHP version 7.2.24 and 7.3.11

### ISSUE THREAD

**Issue by RageZBla** [cloudfoundry/php-buildpack#325]
See [this](https://thehackernews.com/2019/10/nginx-php-fpm-hacking.html) and [that](https://nextcloud.com/blog/urgent-security-issue-in-nginx-php-fpm/)

**Comment by Zanadar** [comment]
@RageZBla These are on develop, if you want to point there. We'll be cutting a release ASAP.

**Comment by dmikusa-pivotal** [comment]
Some other thoughts on this...

1.) By default, your app isn't going to run with Nginx. Most PHP buildpack user will have their app served with Apache HTTPD, so won't be affected by this. If you have not specifically set the web server to Nginx, then you are safe and can stop reading.

2.) To those specifically running Nginx, my understanding from the articles referenced is that to be vulnerable the following need to be true.

>In brief, a website is vulnerable, if:
>
> - NGINX is configured to forward PHP pages requests to PHP-FPM processor,
> - fastcgi_split_path_info directive is present in the configuration and includes a regular expression beginning with a '^' symbol and ending with a '$' symbol,
> - PATH_INFO variable is defined with fastcgi_param directive,
> - There are no checks like try_files $uri =404 or if (-f $uri) to determine whether a file exists or not.

For the default configuration used by the PHP buildpack...

> - NGINX is configured to forward PHP pages requests to PHP-FPM processor,

true

> - fastcgi_split_path_info directive is present in the configuration and includes a regular expression beginning with a '^' symbol and ending with a '$' symbol,

false, I do not see `fastcgi_split_path_info` being used in [our configuration](https://github.com/cloudfoundry/php-buildpack/tree/master/defaults/config/nginx).

> - PATH_INFO variable is defined with fastcgi_param directive,

false, I do not see this being used in [our configuration](https://github.com/cloudfoundry/php-buildpack/tree/master/defaults/config/nginx)

> - There are no checks like try_files $uri =404 or if (-f $uri) to determine whether a file exists or not.

false (which is good in this case). We do include a `try_files` by default before proxying to PHP-FPM. See [here](https://github.com/cloudfoundry/php-buildpack/blob/master/defaults/config/nginx/server-locations.conf#L17).

In summary, I don't think the default configurations are going to be vulnerable to this attack for the reasons able. If you're customizing the configuration, you will need to inspect your customizations.

As always, update as soon as those are available. For reference, that's the following versions:

- 7.1.33
- 7.2.24
- 7.3.11

Hope that helps!

**Comment by Zanadar** [comment]
https://github.com/cloudfoundry/php-buildpack/releases/tag/v4.4.1 has been cut. closing.

**Comment by RageZBla** [comment]
@Zanadar thanks you for the quick turnaround 👍
